### PR TITLE
fix: Whitelist `.MD` and `.github` extensions when checking empty repository

### LINKED ISSUE #11430: [Bug]: Git connect fails when the repo contains yml files
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When the user tries to connect a repo(gitLab or any) contains .yml file connect is failed. `.yml` files are used for CI purpose and we should allow such scenarios.

### Steps To Reproduce

Create a repo in gitlab and choose empty/blank repo
Notice `.yml` CI file added to repo
Now try to connect to the repo and connect fails

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
When user connects to Git, we check whether the repository is empty or not. We allow certain types of files e.g. 
#### PR fixes following issue(s)
Fixes #11430

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Bug Fixes**
	- Enhanced file extension matching to include both lowercase and uppercase extensions, ensuring broader file type support.

### CODE REVIEW

File: app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public class FileUtilsImpl implements FileInterface {
     private static final String VIEW_MODE_URL_TEMPLATE = "{{viewModeUrl}}";
 
     private static final Pattern ALLOWED_FILE_EXTENSION_PATTERN =
-            Pattern.compile("(.*?)\\.(md|git|gitignore|yml|yaml)$");
+            Pattern.compile("(.*?)\\.(md|MD|git|gitignore|github|yml|yaml)$");
```
Comment (sondermanish) [inline app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java:87]: QQ: What is the use of `?` if `*` is there

**Review by sondermanish — APPROVED** [review]
changes LGTM, added auxiliary comment

File: app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java (hunk 1 above)
Comment (nayan-rafiq) [reply app/server/appsmith-git/src/main/java/com/appsmith/git/helpers/FileUtilsImpl.java:87]: @sondermanish I've not checked that. I guess this is to match files with an extension and without a name. For example, for `.gitignore` file, the part before `\\.` will match an optional name. That's why `?` is here.

**Pull request merged**